src/utils/kvMetadata.js

```diff
@@ -0,0 +1,93 @@
+/* global METADATA_BASE */
+
+import fetchJson from './fetchJson.js';
+
+const kvBase = (typeof METADATA_BASE === 'string' ? METADATA_BASE : '') || 'https://metadata.speedcdnjs.com';
+
+/**
+ * Get a list of libraries.
+ *
+ * @return {Promise<string[]>}
+ */
+export const libraries = () => fetchJson(`${kvBase}/packages`);
+
+/**
+ * @template {Object} T
+ *
+ * Validate the data we get from KV for a library.
+ *
+ * @param {string} library Requested library name.
+ * @param {T} data Returned library data to validate.
+ * @param {import('toucan-js')} [sentry] Sentry instance for missing version reporting.
+ * @return {T & { assets: [] }}
+ */
+const kvLibraryValidate = (library, data, sentry = undefined) => {
+    // Assets might not exist if there are none, but we should make it an empty array by default
+    data.assets = data.assets || [];
+
+    // Non-breaking issues
+    if (library !== data.name) {
+        console.info('Name mismatch', library, data.name);
+        data.name = library;
+    }
+
+    // Breaking issues
+    if (!data.version) {
+        console.error('Version missing', data.name, data);
+        if (sentry) {
+            sentry.withScope(scope => {
+                scope.setExtra('data', data);
+                sentry.captureException(new Error('Version missing in package data'));
```

Comment:
If we go with the above suggestion of using `sentry?.withScope`, do we need to use `sentry?.` here? I can't easily find a source for `withScope` so not sure whether the closure called is sync or async. If sync, I assume we don't need `sentry?.`

```suggestion
                sentry?.captureException(new Error('Version missing in package data'));
```